src/persist/src/indexed/trace.rs

```diff
@@ -75,15 +89,21 @@ impl<K: Data, V: Data> BlobTrace<K, V> {
     pub fn meta(&self) -> BlobTraceMeta {
         BlobTraceMeta {
             id: self.id,
-            batches: self.batches.clone(),
+            since: self.since.clone(),
+            // TODO: should we just serialize the level information as well?
```

Comment:
would love a second opinion on this one!

Reply:
Yes. Otherwise we get wildly different behavior after a restart no?

Reply:
(this was a good nerd snipe) I spent a while thinking about this. I think you can kind of hand wave a proof that -- the more batches you have the more likely you are to have a sequence of 1 or two batches at each level

like if you've seen > 1 million batches go by its extremely unlikely that your list of levels will look like

`[20, 1, 1, 1, ,1 ... ]`

and its more likely that your list will look like

`[20, 19, 18, 16, 16, 15, 14, 13, .... 1, 1, 1, ...]`

and if that's true then I can vaguely hand-wave my way into saying "if you recast as 'everything is on the same level' then you wouldn't do more than 2x the optimal amount of work" because intuitively, you want to merge the last three elements when you have

`[20, 19, 19 ... ]`

which turns into 

`[20, 20]`...
 
 which turns into `[21, ... ]`
 
 but in some sense you haven't received enough batches yet to go down the tree to make another level 19, for which you need another level 18, for which you need two more level 17s, ... you need something approximately 2^18 more batches to "pay for" the final level compaction, but the suboptimality of this is (maybe probabilistically) bounded as a fraction over the optimal compaction strategy.
 
 Anyway, that was just a rant; I'll go ahead and serialize the whole thing.